File: src/henaxel/raytracer/materials/NoiseTexture.java
package henaxel.raytracer.materials;

import henaxel.raytracer.node.Node;
import henaxel.raytracer.utils.Vec3;

import java.awt.*;

public class NoiseTexture implements Texture {
    Perlin noise = new Perlin();
    double scale;
    int depth;
    
    public NoiseTexture(double scale, int depth) {
        this.scale = scale;
        this.depth = depth;
    }
    
    @Override
    public Color color(double u, double v, Vec3 hitPos) {
        float perlin = (float)(0.5*(1 + noise.noise(Vec3.multiply(hitPos, scale))));
        float turbulence = (float)(noise.turb(Vec3.multiply(hitPos, scale), depth));
        float marble = (float) (0.5*(1 + Math.sin(scale*hitPos.z + 10*noise.turb(hitPos, depth))));
        return new Color(marble, marble, marble);
    }

    @Override
    public Node getNode() {
        return new Node("Noise Texture", Color.orange, null, null, null, null) {

        };
    }
}

File: src/henaxel/raytracer/materials/NoiseTexture.kt
package henaxel.raytracer.materials

import java.awt.Color
import kotlin.math.sin

class NoiseTexture(var scale: Double, var depth: Int) : Texture() {
    var noise = Perlin()

    override fun color(u: Double, v: Double): Color {
        val perlin = (0.5 * (1 + noise.noise(u * scale, v * scale))).toFloat()
        val turbulence = noise.turb(u * scale, v * scale, depth).toFloat()
        val marble = (0.5 * (1 + sin(scale + 10 * noise.turb(u, v, depth)))).toFloat()
        return Color(marble, marble, marble)
    }
}